samples/perf_client/perf_client.h

```diff
@@ -79,43 +185,48 @@ namespace client
     // check_response for derived-overridable validation
     void process_reply(const RpcTlsClient::Response& reply)
     {
-      if (check_responses)
+      if (options.check_responses)
       {
         if (!check_response(reply))
         {
           throw std::logic_error("Response failed check");
         }
       }
 
-      std::optional<timing::CommitIDs> commits = std::nullopt;
-
-      if (timing.has_value())
+      if (response_times.is_timing_active() && reply.status == HTTP_STATUS_OK)
       {
-        const auto commit_it = reply.headers.find(http::headers::CCF_COMMIT);
-        const auto global_it =
-          reply.headers.find(http::headers::CCF_GLOBAL_COMMIT);
-        const auto term_it = reply.headers.find(http::headers::CCF_TERM);
+        const auto commits = timing::parse_commit_ids(reply);
 
-        // If any of these are missing, we'll write no commits and consider
-        // this a failed request
-        if (
-          commit_it != reply.headers.end() &&
-          global_it != reply.headers.end() && term_it != reply.headers.end())
-        {
-          const size_t commit = std::atoi(commit_it->second.c_str());
-          const size_t global = std::atoi(global_it->second.c_str());
-          const size_t term = std::atoi(term_it->second.c_str());
-          commits.emplace(timing::CommitIDs{commit, global, term});
+        // Record time of received responses
+        response_times.record_receive(reply.id, commits);
 
-          highest_local_commit = std::max<size_t>(highest_local_commit, commit);
+        if (commits.term < last_response_commit.term)
+        {
+          throw std::logic_error(fmt::format(
+            "Term went backwards (expected {}, saw {})!",
+            last_response_commit.term,
+            commits.term));
+        }
+        else if (
+          commits.term > last_response_commit.term &&
+          commits.local <= last_response_commit.index)
+        {
+          throw std::logic_error(fmt::format(
```

Comment:
I guess in the future, if we did want to test across elections as Peng Peng suggested, we would have replay logic here. For now this is an improvement over the current situation already.